tests/validator/test_validator.py

```diff
@@ -645,44 +634,30 @@ def test_graph_validate(in_memory_runtime_context, basic_datasource):
 
 # Tests that runtime configuration actually works during graph validation
 @pytest.mark.big
-def test_graph_validate_with_runtime_config(in_memory_runtime_context, basic_datasource):
-    in_memory_runtime_context.datasources["my_datasource"] = basic_datasource
-    df = pd.DataFrame({"a": [1, 5, 22, 3, 5, 10, 2, 3], "b": [97, 332, 3, 4, 5, 6, 7, None]})
-
-    batch = basic_datasource.get_single_batch_from_batch_request(
-        RuntimeBatchRequest(
-            **{
-                "datasource_name": "my_datasource",
-                "data_connector_name": "test_runtime_data_connector",
-                "data_asset_name": "IN_MEMORY_DATA_ASSET",
-                "runtime_parameters": {
-                    "batch_data": df,
-                },
-                "batch_identifiers": {
-                    "pipeline_stage_name": 0,
-                    "airflow_run_id": 0,
-                    "custom_key_0": 0,
-                },
-            }
-        )
+def test_graph_validate_with_runtime_config(
+    in_memory_runtime_context, basic_datasource: PandasDatasource
+):
+    asset = basic_datasource.add_dataframe_asset(
+        "my_asset",
+        dataframe=pd.DataFrame(
+            {"a": [1, 5, 22, 3, 5, 10, 2, 3], "b": [97, 332, 3, 4, 5, 6, 7, None]}
+        ),
     )
+    batch_definition = asset.add_batch_definition_whole_dataframe("my batch definition")
+    batch = batch_definition.get_batch()
 
     expectation_configuration = ExpectationConfiguration(
         expectation_type="expect_column_value_z_scores_to_be_less_than",
         kwargs={"column": "b", "mostly": 1.0, "threshold": 2.0, "double_sided": True},
     )
-    try:
-        # noinspection PyTypeChecker
-        result = Validator(
-            execution_engine=basic_datasource.execution_engine,
-            data_context=in_memory_runtime_context,
-            batches=[batch],
-        ).graph_validate(
-            configurations=[expectation_configuration],
-            runtime_configuration={"result_format": "COMPLETE"},
-        )
-    except AssertionError as e:
-        result = e
+    result = Validator(
+        execution_engine=PandasExecutionEngine(),
```

Comment:
interesting that this works. it's preferable to accessing a private attr on the datasource, but makes me wonder what our ideal pattern would be.